lib/services/project-changes-service.ts

```diff
@@ -227,10 +228,15 @@ export class ProjectChangesService implements IProjectChangesService {
 
 			let fileStats = this.$fs.getFsStats(filePath);
 
-			let changed = fileStats.mtime.getTime() >= this._outputProjectMtime || fileStats.ctime.getTime() >= this._outputProjectCTime;
-			if (!changed) {
-				let lFileStats = this.$fs.getLsStats(filePath);
-				changed = lFileStats.mtime.getTime() >= this._outputProjectMtime || lFileStats.ctime.getTime() >= this._outputProjectCTime;
+			let changed =  fileStats.mtime.getTime() > this._outputProjectMtime ||
+				fileStats.ctime.getTime() >= this._outputProjectCTime;
+
+			if (this.$hostInfo.isDarwin && !fileStats.isDirectory() && !changed) {
+				const parentDirPath: string = path.dirname(filePath);
+				const parentDirStat = this.$fs.getFsStats(parentDirPath);
```

Comment:
In case we have 50 files in the current dir and nothing is changed (including the parent dir), we'll get the stats of the parent directory 50 times. 
Also, consider the following structure:
```
-- app
   |
   -- components
       |
        -- file1
        -- file 2
```

We'll call `containsNewerFiles` for `app` dir. It will list the contents of the app dir and for each of them it will check the `ctime` and `mtime`, i.e. it will check the times of `components` directory.
In case there's no change, the method will recursively check the contents of the `components` directory. At this point we'll check file1 and file2 and for each of them we'll get the parent dir (`components`) and check its `ctime` and `mtime`. But we've already done this in the previous step, why should we do it again?